devcontainers/setup.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import json
+import typing
+import pathlib
+import argparse
+
+description = """
+This utility script helps you setting up development environment inside docker container
+(called DevContainer), which is quite convenient for versioning and well supported by VSCode.
+
+Generally you only provide setup name, which is displayed as config name in VSCode (-s, --setup-name), and
+distribution which you wish to work with. You can also specify output json filepath, individual component versions
+(the ones compiled in container, anyways) and user name to use inside the container.
+
+Before setting versions manually, check github repos of respective component for correct tag.
+
+Example:
+./devcontainers/setup.py -s example-config -d arch-linux --output .devcontainer/example.json
+"""
+
+DEVCONTAINER_CONFIG_PATH = '.devcontainer/devcontainer.json'
```

Comment:
The *.devcontainer* directory needs to be created by the user explicitly, i.e. this script fails with its default settings. The customizations offered by *setup.py* can be realized via plain Docker arguments controlled by these JSON files. Do I miss an advantage?